### ISSUE TITLE
Enum constructor accepts signed int

### ISSUE THREAD

**Issue by axic** [ethereum/solidity#1343]
```
contract C {
    enum ActionChoices { GoLeft, GoRight, GoStraight }
	function getChoice() returns (int d)
	{
	    return int(ActionChoices(-1));
	}
}
```

Returns `-1`, which is invalid enum value.

**Comment by axic** [comment]
@pirapira probably this could be fixed by #1334 easily?

**Comment by pirapira** [comment]
@axic I added a test in #1334.  For `-1` #1334 already works.
